Observer/ControllerActionPredispatchObserver.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace JustBetter\Sentry\Observer;
+
+use JustBetter\Sentry\Model\SentryPerformance;
+use Magento\Framework\App\Request\Http;
+use Magento\Framework\Event\ObserverInterface;
+
+class ControllerActionPredispatchObserver implements ObserverInterface
+{
+    /** @var SentryPerformance */
+    private $sentryPerformance;
+
+    /** @var Http */
+    private $request;
+
+    public function __construct(SentryPerformance $sentryPerformance, Http $request)
+    {
+        $this->sentryPerformance = $sentryPerformance;
+        $this->request = $request;
+    }
+
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $this->sentryPerformance->startTransaction($this->request);
```

Comment:
Might it be a good idea to add the check wether we're doing performance tracking in these places as well? To ensure no code related to performance tracking is executed when it's disabled